### ISSUE TITLE
CJK support

### ISSUE THREAD

**Issue by BornForFever** [kivy/kivy#6790]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when kivy doesn't support cjk. When I set a widget's `text` to Chinese, the text will be actually shown in boxes.

**Describe the solution you'd like**
Add CJK fonts and a fallback mechanism

**Describe alternatives you've considered**
set `font_path` to a cjk font

**Additional context**
none

**Comment by tshirtman** [comment]
what's preventing you to use a font that has chinese characters in it? I agree font_fallback is the best solution, and we have it with the pango text provider, but it's linux only, quite some work is needed to implement it for the others (if even possible). While setting the font_name property of your label works, or is there something with that that doesn't work?

**Comment by BornForFever** [comment]
It’s just problematic.

**Comment by matham** [comment]
Is this asking to include a default font that supports cjk? If so, I'm guessing PRs are welcome?

**Comment by tshirtman** [comment]
that would sensibly increase the size of the apps, as fonts supporting all international characters tend to weight in the megabytes, if that was included by default, it would make all the apks packages that much bigger. I think it's better to simply document how to use and include a font that supports international characters.

**Comment by BornForFever** [comment]
Yes. But I'm worring that with users from more and more countries using kivy, kivy will have to support increasing languages, making kivy become bloated. Hmmmm....

**Comment by tshirtman** [comment]
alternatively if there is a way when rendering a text that the font doesn't support some of the character rendered, it would be possible to display a warning indicating the user what steps to take to allow the app displaying them.

**Comment by matham** [comment]
I don't like adding support for a popup, because that would insert ourselves too much into the control flow of user apps. To even make it work we'd have to make judgements about how to show this and when to show this - that can become hacky quickly and is much better left to app developers.

I think a better solution is to bundle all the fonts into `kivy-fonts-extra` package and then add this package as a extra dependency named `fonts` so when someone installs it they just specify `pip install kivy[fonts]` and it works.

As an aside, one of the paths forward that I think kivy needs to take, is to break up kivy into some of its dependencies (e.g. `properties` and `EventDispatcher` should be a external `kivy-dispatch` package that kivy depends on). This would make maintenance much better as people could focus on bite-sized projects that are self-contained and cleaner, and also other people could use these sub-projects and contribute to them without having to use the whole kivy. That would increase our user-base and contributors without adding yet more code to kivy. I've seen other projects do this to success. This may make issues more manageable also as it would be broken down into relevant projects rather than some generic issue among 700 in the main kivy repo.

**Comment by tshirtman** [comment]
oh by warning i meant in the logs, and for the benefit of the developer, sorry i wasn't clear. I'm not sure we even *can* detect text that is not rendered properly by the core provider, so that's hypothetical.

Would the [fonts] option work for example on python-for-android?

I think the remaining part of the message is interesting but should rather be in another issue :).

**Comment by matham** [comment]
Yeah, a warning if we can detect this would seem appropriate. Although wouldn't the problem be immediately obvious as the text is not rendered, so a logged warning would not really add much? Except I guess to say to install `fonts`.

>Would the [fonts] option work for example on python-for-android?

If it doesn't, it should be fixed to support it because that's a generic pip option commonly used.
